core/base/ftmTree/FTMTree_MT.h

```diff
@@ -642,6 +657,29 @@ namespace ttk {
       void getTreeBranching(std::vector<idNode> &branching,
                             std::vector<int> &branchingID);
 
+      std::vector<idNode> getAllRoots();
+
+      int getNumberOfRoot();
+
+      int getNumberOfChildren(idNode nodeId);
+
+      int getTreeDepth();
+
+      int getNodeLevel(idNode nodeId);
+
+      std::vector<int> getAllNodeLevel();
+
+      std::vector<std::vector<idNode>> getLevelToNode();
```

Comment:
please do not return objects (especially containers).
instead pass the object to fill (i.e. the output) by reference as a parameter of the function.
(that will avoid time copies).

please address this concern throughout your entire code (including the merged PR #633 if need be).

Reply:
Thanks! Done in the next commit!